python/private/py_wheel.bzl

```diff
@@ -238,8 +275,8 @@ def _py_wheel_impl(ctx):
     version = _replace_make_variables(ctx.attr.version, ctx)
 
     outfile = ctx.actions.declare_file("-".join([
-        _escape_filename_segment(ctx.attr.distribution),
-        _escape_filename_segment(version),
+        _escape_filename_distribution_name(ctx.attr.distribution),
+        normalize_pep440(version),
```

Comment:
It would be great if we could make this new behaviour optional and gated under a boolean flag so that users that depend on the broken behaviour have more time to migrate.

The plan could be:
* Add a flag named `normalize_version_pep440` which defaults to `False` for now.
* If the flag is set to `True` then the new codepath is used.
* In a future release the `incompatible_normalize_version` is set to `True` but users have ability to set it to False if required.

That way the change is no longer breaking.

Consider adding an extra dictionary with args named:
```starlark
_feature_flags = {
    "incompatible_normalize_version": attr.bool(
        default = False,
        doc = "Normalize the package version according to PEP440 standard",
    ),
    "incompatible_explicit_distribution_stamp_vars": attr.bool(
        default = False,
        doc = "Change the format of how the stamp variables can be used to " +
              "in the distribution name. With this option set to True, the " +
              "user has to pass stamp variables enclosed in '{}', e.g. " +
              "'{BUILD_TIMESTAMP}'."
    ),
}
```

Reply:
To add some context (I was out and about on my phone when Ignas asked me about this, so I didn't have time to respond in full):

The key point is to decouple upgrading the rules_python version from having to address incompatible changes. This allows a more incremental and manageable upgrade process. A user can upgrade rules_python, opt into behavior where they can, and followup later where they can't. The next version (which has the incompatible behavior by default) blocks them from upgrading, but they're still able to make their current version forward compatible.